internal/ui/dialog/delete.go

```diff
@@ -4,27 +4,36 @@ import (
 	"github.com/derailed/k9s/internal/config"
 	"github.com/derailed/k9s/internal/ui"
 	"github.com/derailed/tview"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const deleteKey = "delete"
 
+const defaultPropagationIdx = 0
+
+var propagationOptions []string = []string{
+	string(metav1.DeletePropagationBackground),
+	string(metav1.DeletePropagationForeground),
+	string(metav1.DeletePropagationOrphan),
+}
+
 type (
-	okFunc     func(cascade, force bool)
+	okFunc     func(propagation string, force bool)
 	cancelFunc func()
 )
 
 // ShowDelete pops a resource deletion dialog.
 func ShowDelete(styles config.Dialog, pages *ui.Pages, msg string, ok okFunc, cancel cancelFunc) {
-	cascade, force := true, false
+	propagation, force := "", false
 	f := tview.NewForm()
 	f.SetItemPadding(0)
 	f.SetButtonsAlign(tview.AlignCenter).
 		SetButtonBackgroundColor(styles.ButtonBgColor.Color()).
 		SetButtonTextColor(styles.ButtonFgColor.Color()).
 		SetLabelColor(styles.LabelFgColor.Color()).
 		SetFieldTextColor(styles.FieldFgColor.Color())
-	f.AddCheckbox("Cascade:", cascade, func(_ string, checked bool) {
-		cascade = checked
+	f.AddDropDown("Propagation:", propagationOptions, defaultPropagationIdx, func(_ string, optionIndex int) {
```

Comment:
Think we should have a "none" option here as the policy is optional and can be set to nil which will used the existing finalizer if unset